synthtool/languages/python.py

```diff
@@ -93,6 +94,36 @@ def _get_sample_readme_metadata(sample_dir: Path) -> dict:
     return sample_metadata
 
 
+def detect_versions(
+    path: str = "./src", default_version: Optional[str] = None
+) -> List[str]:
+    """
+    Detects the versions a library has, based on distinct folders
+    within path. This is based on the fact that our GAPIC libraries are
+    structured as follows:
+
+    src/v1
+    src/v1beta
+    src/v1alpha
+
+    With folder names mapping directly to versions.
+
+    Returns: a list of the subdirectories; for the example above:
+      ['v1', 'v1alpha', 'v1beta']
+      If specified, the default_version is guaranteed to be listed last.
+      Otherwise, the list is sorted alphabetically.
+    """
+    versions = []
+    if os.path.isdir(path):
+        for directory in os.listdir(path):
+            if os.path.isdir(os.path.join(path, directory)):
+                versions.append(directory)
+    versions.sort()
+    if default_version is not None:
+        versions = [v for v in versions if v != default_version] + [default_version]
```

Comment:
Can `default_version` be put first in the list? I think that's the pattern currently used in the docs. https://googleapis.dev/python/automl/latest/index.html

Reply:
Done in https://github.com/googleapis/synthtool/pull/1114/commits/89e2bc43056196d5c537d8ac5298c269c0c52eb6